## 1. Importing Necessary Libraries

In [2]:
import os

import numpy as np

import pandas as pd

from sklearn.model_selection import train_test_split

## 2. Load Data

In [4]:
PROJECT_DIR = r'/Users/bakhtsinghbasaram/Downloads/flights-sagemaker-project'
DATA_DIR = 'data'

def get_data(filename):
    file_name = f"{filename}.csv"
    file_path = os.path.join(PROJECT_DIR, DATA_DIR, file_name)
    return pd.read_csv(file_path)


In [5]:
flights = get_data("flight_price")
flights

Airline Date_of_Journey    Source Destination  \
0           IndiGo      24/03/2019  Banglore   New Delhi   
1        Air India       1/05/2019   Kolkata    Banglore   
2      Jet Airways       9/06/2019     Delhi      Cochin   
3           IndiGo      12/05/2019   Kolkata    Banglore   
4           IndiGo      01/03/2019  Banglore   New Delhi   
...            ...             ...       ...         ...   
10678     Air Asia       9/04/2019   Kolkata    Banglore   
10679    Air India      27/04/2019   Kolkata    Banglore   
10680  Jet Airways      27/04/2019  Banglore       Delhi   
10681      Vistara      01/03/2019  Banglore   New Delhi   
10682    Air India       9/05/2019     Delhi      Cochin   

                       Route Dep_Time  Arrival_Time Duration Total_Stops  \
0                  BLR → DEL    22:20  01:10 22 Mar   2h 50m    non-stop   
1      CCU → IXR → BBI → BLR    05:50         13:15   7h 25m     2 stops   
2      DEL → LKO → BOM → COK    09:25  04:25 10 Jun      19h     2 stops   
3            CCU → NAG → BLR    18:05         23:30   5h 25m      1 stop   
4            BLR → NAG → DEL    16:50         21:35   4h 45m      1 stop   
...                      ...      ...           ...      ...         ...   
10678              CCU → BLR    19:55         22:25   2h 30m    non-stop   
10679              CCU → BLR    20:45         23:20   2h 35m    non-stop   
10680              BLR → DEL    08:20         11:20       3h    non-stop   
10681              BLR → DEL    11:30         14:10   2h 40m    non-stop   
10682  DEL → GOI → BOM → COK    10:55         19:15   8h 20m     2 stops   

      Additional_Info  Price  
0             No info   3897  
1             No info   7662  
2             No info  13882  
3             No info   6218  
4             No info  13302  
...               ...    ...  
10678         No info   4107  
10679         No info   4145  
10680         No info   7229  
10681         No info  12648  
10682         No info  11753  

[10683 rows x 11 columns]

In [6]:
# get basic idea of the dataset
flights.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10683 entries, 0 to 10682
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Airline          10683 non-null  object
 1   Date_of_Journey  10683 non-null  object
 2   Source           10683 non-null  object
 3   Destination      10683 non-null  object
 4   Route            10682 non-null  object
 5   Dep_Time         10683 non-null  object
 6   Arrival_Time     10683 non-null  object
 7   Duration         10683 non-null  object
 8   Total_Stops      10682 non-null  object
 9   Additional_Info  10683 non-null  object
 10  Price            10683 non-null  int64 
dtypes: int64(1), object(10)
memory usage: 918.2+ KB


- The dataset contains 10,683 rows and 11 features
- Columns Route and Total_Stops have missing value each
- The data types of some features isn't appropriate. Ex: Date_of_Journey should be in "datetime" but it is of "object" type

## 3. Preliminary Analysis

### 3.1 Check Data Types

In [10]:
# view first 7 rows of dataframe
flights.head(7)

Airline Date_of_Journey    Source Destination                  Route  \
0       IndiGo      24/03/2019  Banglore   New Delhi              BLR → DEL   
1    Air India       1/05/2019   Kolkata    Banglore  CCU → IXR → BBI → BLR   
2  Jet Airways       9/06/2019     Delhi      Cochin  DEL → LKO → BOM → COK   
3       IndiGo      12/05/2019   Kolkata    Banglore        CCU → NAG → BLR   
4       IndiGo      01/03/2019  Banglore   New Delhi        BLR → NAG → DEL   
5     SpiceJet      24/06/2019   Kolkata    Banglore              CCU → BLR   
6  Jet Airways      12/03/2019  Banglore   New Delhi        BLR → BOM → DEL   

  Dep_Time  Arrival_Time Duration Total_Stops              Additional_Info  \
0    22:20  01:10 22 Mar   2h 50m    non-stop                      No info   
1    05:50         13:15   7h 25m     2 stops                      No info   
2    09:25  04:25 10 Jun      19h     2 stops                      No info   
3    18:05         23:30   5h 25m      1 stop                      No info   
4    16:50         21:35   4h 45m      1 stop                      No info   
5    09:00         11:25   2h 25m    non-stop                      No info   
6    18:55  10:25 13 Mar  15h 30m      1 stop  In-flight meal not included   

   Price  
0   3897  
1   7662  
2  13882  
3   6218  
4  13302  
5   3873  
6  11087

In [11]:
# view the data types of all columns
flights.dtypes

Airline            object
Date_of_Journey    object
Source             object
Destination        object
Route              object
Dep_Time           object
Arrival_Time       object
Duration           object
Total_Stops        object
Additional_Info    object
Price               int64
dtype: object

In [12]:
flights.Date_of_Journey.iloc[2]

'9/06/2019'

In [13]:
flights.Dep_Time.iloc[100]

'08:25'

### 3.2 Check for Duplicates

In [15]:
# want to the total number of duplicates
print(flights.duplicated().sum())

220


In [16]:
# display duplicate rows for better understanding

(
    flights
    .loc[flights.duplicated(keep=False)]
    .sort_values(["Airline", "Date_of_Journey", "Source", "Destination"])
)

#Note:
#flights.duplicated() (default keep='first'): Marks all duplicates as True except for the first occurrence.
#flights.duplicated(keep=False): Marks all duplicates as True, including all occurrences.

Airline Date_of_Journey    Source Destination                  Route  \
6321  Air India      01/03/2019  Banglore   New Delhi  BLR → BOM → AMD → DEL   
9848  Air India      01/03/2019  Banglore   New Delhi  BLR → BOM → AMD → DEL   
572   Air India      03/03/2019  Banglore   New Delhi              BLR → DEL   
8168  Air India      03/03/2019  Banglore   New Delhi              BLR → DEL   
1495  Air India       1/04/2019   Kolkata    Banglore  CCU → DEL → COK → BLR   
...         ...             ...       ...         ...                    ...   
2692   SpiceJet      24/03/2019  Banglore   New Delhi              BLR → DEL   
2870   SpiceJet      24/03/2019  Banglore   New Delhi              BLR → DEL   
3711   SpiceJet      24/03/2019  Banglore   New Delhi              BLR → DEL   
2634    Vistara      24/03/2019  Banglore   New Delhi              BLR → DEL   
5470    Vistara      24/03/2019  Banglore   New Delhi              BLR → DEL   

     Dep_Time  Arrival_Time Duration Total_Stops  \
6321    08:50  23:55 02 Mar   39h 5m     2 stops   
9848    08:50  23:55 02 Mar   39h 5m     2 stops   
572     21:10         23:55   2h 45m    non-stop   
8168    21:10         23:55   2h 45m    non-stop   
1495    10:00  01:20 02 Apr  15h 20m     2 stops   
...       ...           ...      ...         ...   
2692    05:45         08:35   2h 50m    non-stop   
2870    05:45         08:35   2h 50m    non-stop   
3711    20:30         23:20   2h 50m    non-stop   
2634    11:30         14:10   2h 40m    non-stop   
5470    11:30         14:10   2h 40m    non-stop   

                   Additional_Info  Price  
6321                       No info  17135  
9848                       No info  17135  
572                        No info   7591  
8168                       No info   7591  
1495                       No info  10408  
...                            ...    ...  
2692  No check-in baggage included   4273  
2870  No check-in baggage included   4273  
3711  No check-in baggage included   3873  
2634                       No info   5403  
5470                       No info   5403  

[415 rows x 11 columns]

### 3.3 Observations

- The type of Date_of_Journey, Dep_Time and Arrival_Time should be changed to datetime
- The type of Duration and Total_Stops is mixed(you see numbers and strings together). It should be numeric type.
- There are 220 duplicates and these should be removed.

## 4. Detailed Analysis

In [20]:
# Let's view Airline Column
flights.Airline

0             IndiGo
1          Air India
2        Jet Airways
3             IndiGo
4             IndiGo
            ...     
10678       Air Asia
10679      Air India
10680    Jet Airways
10681        Vistara
10682      Air India
Name: Airline, Length: 10683, dtype: object

In [21]:
# check the unique values in Airline column
flights.Airline.unique()

array(['IndiGo', 'Air India', 'Jet Airways', 'SpiceJet',
       'Multiple carriers', 'GoAir', 'Vistara', 'Air Asia',
       'Vistara Premium economy', 'Jet Airways Business',
       'Multiple carriers Premium economy', 'Trujet'], dtype=object)

- Vistara and Vistara Premium economy doesn't seem to different airways. They should belong to same airways but may be of different classes.
- Same applies with Jet Airways.
- let's fix these inconsistencies in the data

In [23]:
(
    flights
    .Airline
    .str.replace(" Premium economy", "")
	.str.replace(" Business", "")
    .str.title() # convert the first character of each word in a Series of strings to uppercase and the remaining characters to lowercase
)

0             Indigo
1          Air India
2        Jet Airways
3             Indigo
4             Indigo
            ...     
10678       Air Asia
10679      Air India
10680    Jet Airways
10681        Vistara
10682      Air India
Name: Airline, Length: 10683, dtype: object

#### Date_of_Journey

In [25]:
flights.Date_of_Journey.head()

0    24/03/2019
1     1/05/2019
2     9/06/2019
3    12/05/2019
4    01/03/2019
Name: Date_of_Journey, dtype: object

In [26]:
pd.to_datetime(flights.Date_of_Journey, dayfirst=True)

#Using pd.to_datetime(flights.Date_of_Journey, dayfirst=True) ensures that the date strings are correctly parsed into datetime objects 
#with the day coming first. This is especially important for avoiding errors and ensuring that dates are interpreted correctly 
#according to the intended format.

0       2019-03-24
1       2019-05-01
2       2019-06-09
3       2019-05-12
4       2019-03-01
           ...    
10678   2019-04-09
10679   2019-04-27
10680   2019-04-27
10681   2019-03-01
10682   2019-05-09
Name: Date_of_Journey, Length: 10683, dtype: datetime64[ns]

#### Source

In [28]:
flights.Source.unique()

array(['Banglore', 'Kolkata', 'Delhi', 'Chennai', 'Mumbai'], dtype=object)

- Banglore is misspelled but it is fine in this case. We would only deal with that if there are two different unique values like 'Banglore' and 'Bangalore'

#### Destination

In [31]:
flights.Destination.unique()

array(['New Delhi', 'Banglore', 'Cochin', 'Kolkata', 'Delhi', 'Hyderabad'],
      dtype=object)

#### Dep_Time

In [33]:
flights.Dep_Time.head()

0    22:20
1    05:50
2    09:25
3    18:05
4    16:50
Name: Dep_Time, dtype: object

In [34]:
# Let's see if there are any values that doesn't start with integer and colon.

(
	flights
	.Dep_Time
	.loc[lambda ser: ser.str.contains("[^0-9:]")]
)

Series([], Name: Dep_Time, dtype: object)

In [35]:
# pd.to_datetime(flights.Dep_Time).dt.time is used to convert a column of time-like strings in a pandas DataFrame 
# into Python datetime.time objects, extracting only the time component from each datetime.

pd.to_datetime(flights.Dep_Time).dt.time

/var/folders/pv/7lpnwxrx1gd70gtxv_gv0_y80000gn/T/ipykernel_17785/3625779338.py:4: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  pd.to_datetime(flights.Dep_Time).dt.time


0        22:20:00
1        05:50:00
2        09:25:00
3        18:05:00
4        16:50:00
           ...   
10678    19:55:00
10679    20:45:00
10680    08:20:00
10681    11:30:00
10682    10:55:00
Name: Dep_Time, Length: 10683, dtype: object

#### Arrival_Time

In [37]:
flights.Arrival_Time

0        01:10 22 Mar
1               13:15
2        04:25 10 Jun
3               23:30
4               21:35
             ...     
10678           22:25
10679           23:20
10680           11:20
10681           14:10
10682           19:15
Name: Arrival_Time, Length: 10683, dtype: object

In [38]:
(
	flights
	.Arrival_Time
	.loc[lambda ser: ser.str.contains("[^0-9:]")] # Check if there are any values that doesn't start with integer and colon.
	.str.split(" ", n=1) # splits 01:10 22 Mar to [01:10, 22 Mar]
	.str.get(1) # gets value of index 1 from [01:10, 22 Mar]
	.unique() # get unique values from the entire series of strings
)

array(['22 Mar', '10 Jun', '13 Mar', '02 Mar', '10 May', '04 Mar',
       '13 Jun', '28 May', '19 Mar', '07 May', '02 Jun', '16 Jun',
       '19 May', '16 May', '28 Jun', '02 May', '28 Mar', '19 Jun',
       '04 Apr', '25 Mar', '07 Mar', '25 Jun', '07 Jun', '25 May',
       '13 May', '16 Mar', '22 May', '10 Apr', '04 Jun', '20 May',
       '28 Apr', '25 Apr', '10 Mar', '19 Apr', '13 Apr', '02 Apr',
       '23 Mar', '22 Apr', '11 May', '07 Apr', '03 May', '08 Mar',
       '03 Mar', '05 Mar', '22 Jun', '04 May', '26 May', '16 Apr',
       '26 Jun', '29 May', '29 Jun', '29 Mar', '23 May', '17 Jun'],
      dtype=object)

#### Duration

In [40]:
flights.Duration

0        2h 50m
1        7h 25m
2           19h
3        5h 25m
4        4h 45m
          ...  
10678    2h 30m
10679    2h 35m
10680        3h
10681    2h 40m
10682    8h 20m
Name: Duration, Length: 10683, dtype: object

In [41]:
(
	flights
	.Duration
	.loc[lambda ser: ~ser.str.contains("m")] # # Check if there are any values that doesn't contain 'm'.
	.unique()
)

array(['19h', '23h', '22h', '12h', '3h', '5h', '10h', '18h', '24h', '15h',
       '16h', '8h', '14h', '20h', '13h', '11h', '9h', '27h', '26h', '4h',
       '7h', '30h', '21h', '28h', '47h', '6h', '25h', '38h', '34h'],
      dtype=object)

In [42]:
(
	flights
	.Duration
	.loc[lambda ser: ~ser.str.contains("h")] # # Check if there are any values that doesn't contain 'h'.
	.unique()
)

array(['5m'], dtype=object)

- We can't imagine a flight trip that has duration of only 5 minutes. Let's find the index of this row and investigate more

In [44]:
(
	flights
	.Duration
	.loc[lambda ser: ~ser.str.contains("h")] # # Check if there are any values that doesn't contain 'h'.
)

6474    5m
Name: Duration, dtype: object

In [45]:
# display index 6474

flights.iloc[[6474]]

Airline Date_of_Journey  Source Destination                  Route  \
6474  Air India       6/03/2019  Mumbai   Hyderabad  BOM → GOI → PNQ → HYD   

     Dep_Time Arrival_Time Duration Total_Stops Additional_Info  Price  
6474    16:50        16:55       5m     2 stops         No info  17327

- 2 stops in duration of 5 minutes is not realistic. If we have more values of similar kind we can use some imputation methods. Because we have only 1 value we can drop this row.

In [47]:
# Convert Duration column into minutes. Instead of Hours and Minutes

(
	flights
	.Duration #  2h 50m
	.drop(index=6474)
	.str.split(" ", expand=True) # [2h, 50m]
	.set_axis(["hour", "minute"], axis=1) # temp columns with names 'hour' and 'minute'
	.assign(
		hour=lambda df_: (
			df_
			.hour 
			.str.replace("h", "") # replace value of "hour" column for example 2h into 2
			.astype(int) # convert object type of 2 -> integer type
			.mul(60) # multiply that with minutes 
		),
		minute=lambda df_: (
			df_
			.minute
			.str.replace("m", "") # replace value of "minute" column for example 50m into 50
			.fillna("0") # Nan values are replaced with 0
			.astype(int) # change the value to int type
		)
	)
	.sum(axis=1) # sum them column wise. Ex: 120 + 50 = 170
)

# Hint: The .assign() function in pandas is used to add new columns to a DataFrame or modify existing ones. 
#This function returns a new DataFrame, leaving the original DataFrame unchanged. 
#The .assign() function is particularly useful for chaining operations together in a readable and concise manner.

0         170
1         445
2        1140
3         325
4         285
         ... 
10678     150
10679     155
10680     180
10681     160
10682     500
Length: 10682, dtype: int64

In [48]:
# compare the above calculated results with Duration column by placing it side by side

(
	flights
	.Duration
	.drop(index=6474)
	.str.split(" ", expand=True)
	.set_axis(["hour", "minute"], axis=1)
	.assign(
		hour=lambda df_: (
			df_
			.hour
			.str.replace("h", "")
			.astype(int)
			.mul(60)
		),
		minute=lambda df_: (
			df_
			.minute
			.str.replace("m", "")
			.fillna("0")
			.astype(int)
		)
	)
	.sum(axis=1)
	.rename("duration_minutes")
	.to_frame()
	.join(flights.Duration.drop(index=6474))
)

duration_minutes Duration
0                   170   2h 50m
1                   445   7h 25m
2                  1140      19h
3                   325   5h 25m
4                   285   4h 45m
...                 ...      ...
10678               150   2h 30m
10679               155   2h 35m
10680               180       3h
10681               160   2h 40m
10682               500   8h 20m

[10682 rows x 2 columns]

#### Total_Stops

In [50]:
flights.Total_Stops.unique()

array(['non-stop', '2 stops', '1 stop', '3 stops', nan, '4 stops'],
      dtype=object)

In [51]:
# Let's Convert column "Total_Stops" from Obj type to Int

(
	flights
	.Total_Stops
	.replace("non-stop", "0")
	.str.replace(" stops?", "", regex=True) # '?' after 's' means, it is optional (stop, stops) 
	.pipe(lambda ser: pd.to_numeric(ser)) # .astype(int) will not work here as we have missing values
)

0        0.0
1        2.0
2        2.0
3        1.0
4        1.0
        ... 
10678    0.0
10679    0.0
10680    0.0
10681    0.0
10682    2.0
Name: Total_Stops, Length: 10683, dtype: float64

#### Additional_Info

In [53]:
flights.Additional_Info.unique()

array(['No info', 'In-flight meal not included',
       'No check-in baggage included', '1 Short layover', 'No Info',
       '1 Long layover', 'Change airports', 'Business class',
       'Red-eye flight', '2 Long layover'], dtype=object)

#### Now we will make all the above operations that we practiced into functions

## 5. Cleaning Operations

In [56]:
def convert_to_minutes(ser):
	return (
		ser
		.str.split(" ", expand=True)
		.set_axis(["hour", "minute"], axis=1)
		.assign(
			hour=lambda df_: (
				df_
				.hour
				.str.replace("h", "")
				.astype(int)
				.mul(60)
			),
			minute=lambda df_: (
				df_
				.minute
				.str.replace("m", "")
				.fillna("0")
				.astype(int)
			)
		)
		.sum(axis=1)
	)

In [57]:
def clean_data(df):
	return (
		df
		.drop(index=[6474])
		.drop_duplicates()
		.assign(**{
			col: df[col].str.strip()
			for col in df.select_dtypes(include="O").columns
		})
		.rename(columns=str.lower)
		.assign(
			airline=lambda df_: (
				df_
				.airline
				.str.replace(" Premium economy", "")
				.str.replace(" Business", "")
				.str.title()
			),
			date_of_journey=lambda df_: pd.to_datetime(df_.date_of_journey, dayfirst=True),
			dep_time=lambda df_: pd.to_datetime(df_.dep_time).dt.time,
			arrival_time=lambda df_: pd.to_datetime(df_.arrival_time).dt.time,
			duration=lambda df_: df_.duration.pipe(convert_to_minutes),
			total_stops=lambda df_: (
				df_
				.total_stops
				.replace("non-stop", "0")
				.str.replace(" stops?", "", regex=True)
				.pipe(lambda ser: pd.to_numeric(ser))
			),
			additional_info=lambda df_: df_.additional_info.replace("No info", "No Info")
		)
		.drop(columns="route")
	)

In [58]:
flights_cleaned = clean_data(flights)
flights_cleaned

/var/folders/pv/7lpnwxrx1gd70gtxv_gv0_y80000gn/T/ipykernel_17785/3708629844.py:20: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  dep_time=lambda df_: pd.to_datetime(df_.dep_time).dt.time,
/var/folders/pv/7lpnwxrx1gd70gtxv_gv0_y80000gn/T/ipykernel_17785/3708629844.py:21: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  arrival_time=lambda df_: pd.to_datetime(df_.arrival_time).dt.time,


airline date_of_journey    source destination  dep_time  \
0           Indigo      2019-03-24  Banglore   New Delhi  22:20:00   
1        Air India      2019-05-01   Kolkata    Banglore  05:50:00   
2      Jet Airways      2019-06-09     Delhi      Cochin  09:25:00   
3           Indigo      2019-05-12   Kolkata    Banglore  18:05:00   
4           Indigo      2019-03-01  Banglore   New Delhi  16:50:00   
...            ...             ...       ...         ...       ...   
10678     Air Asia      2019-04-09   Kolkata    Banglore  19:55:00   
10679    Air India      2019-04-27   Kolkata    Banglore  20:45:00   
10680  Jet Airways      2019-04-27  Banglore       Delhi  08:20:00   
10681      Vistara      2019-03-01  Banglore   New Delhi  11:30:00   
10682    Air India      2019-05-09     Delhi      Cochin  10:55:00   

      arrival_time  duration  total_stops additional_info  price  
0         01:10:00       170          0.0         No Info   3897  
1         13:15:00       445          2.0         No Info   7662  
2         04:25:00      1140          2.0         No Info  13882  
3         23:30:00       325          1.0         No Info   6218  
4         21:35:00       285          1.0         No Info  13302  
...            ...       ...          ...             ...    ...  
10678     22:25:00       150          0.0         No Info   4107  
10679     23:20:00       155          0.0         No Info   4145  
10680     11:20:00       180          0.0         No Info   7229  
10681     14:10:00       160          0.0         No Info  12648  
10682     19:15:00       500          2.0         No Info  11753  

[10462 rows x 10 columns]

## 6. Split the Data

In [60]:
#flights_final = flights_cleaned.sample(1000) run this cell only if you want to play with AWS Sagemaker, 
#if data is more then execution time is more
# hence there will be more charges

In [61]:
X = flights_cleaned.drop(columns="price")
y = flights_cleaned.price.copy()

In [62]:
X_, X_test, y_, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
X_train, X_val, y_train, y_val = train_test_split(X_, y_, test_size=0.2, random_state=42)

print(X_train.shape, y_train.shape)
print(X_val.shape, y_val.shape)
print(X_test.shape, y_test.shape)

(6695, 9) (6695,)
(1674, 9) (1674,)
(2093, 9) (2093,)


In [63]:
def export_data(X, y, name):
	file_name = f"{name}.csv"
	file_path = os.path.join(PROJECT_DIR, DATA_DIR, file_name)

	X.join(y).to_csv(file_path, index=False)

	return pd.read_csv(file_path).head()

In [64]:
export_data(X_train, y_train, "train")

airline date_of_journey    source destination  dep_time arrival_time  \
0  Jet Airways      2019-03-21  Banglore   New Delhi  08:55:00     19:10:00   
1  Jet Airways      2019-03-27     Delhi      Cochin  17:30:00     04:25:00   
2        Goair      2019-03-09  Banglore   New Delhi  11:40:00     14:35:00   
3    Air India      2019-06-12   Kolkata    Banglore  09:25:00     18:30:00   
4  Jet Airways      2019-03-12  Banglore   New Delhi  22:55:00     07:40:00   

   duration  total_stops              additional_info  price  
0       615          1.0  In-flight meal not included   7832  
1       655          1.0  In-flight meal not included   6540  
2       175          0.0                      No Info   7305  
3       545          1.0                      No Info   8366  
4       525          1.0  In-flight meal not included  11087

In [65]:
export_data(X_val, y_val, "val")

airline date_of_journey    source destination  dep_time  \
0             Indigo      2019-06-24     Delhi      Cochin  20:25:00   
1  Multiple Carriers      2019-06-12     Delhi      Cochin  09:45:00   
2        Jet Airways      2019-03-12  Banglore   New Delhi  22:55:00   
3  Multiple Carriers      2019-06-06     Delhi      Cochin  13:00:00   
4        Jet Airways      2019-05-18     Delhi      Cochin  23:05:00   

  arrival_time  duration  total_stops              additional_info  price  
0     01:30:00       305          1.0                      No Info   5054  
1     22:30:00       765          1.0                      No Info   9646  
2     15:15:00       980          1.0  In-flight meal not included  11087  
3     21:00:00       480          1.0                      No Info  13587  
4     04:25:00      1760          2.0                      No Info  16704

In [66]:
export_data(X_test, y_test, "test")

airline date_of_journey    source destination  dep_time arrival_time  \
0  Jet Airways      2019-03-06  Banglore   New Delhi  08:00:00     08:15:00   
1     Spicejet      2019-06-06   Kolkata    Banglore  22:20:00     00:40:00   
2       Indigo      2019-03-18   Kolkata    Banglore  05:30:00     08:20:00   
3       Indigo      2019-06-27   Chennai     Kolkata  19:35:00     21:55:00   
4       Indigo      2019-05-06   Kolkata    Banglore  15:15:00     17:45:00   

   duration  total_stops additional_info  price  
0      1455          1.0         No Info  17996  
1       140          0.0         No Info   3873  
2       170          0.0         No Info   4462  
3       140          0.0         No Info   3597  
4       150          0.0         No Info   4804